server/src/carto/carto.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import * as fetch from 'node-fetch';
+
+@Injectable()
+export class CartoService {
+  cartoUsername = 'planninglabs';
+
+  buildSqlUrl(cleanedQuery, format = 'json', method) { // eslint-disable-line
+    let url = `https://${this.cartoUsername}.carto.com/api/v2/sql`;
+    url += method === 'get' ? `?q=${cleanedQuery}&format=${format}` : '';
+
+    return url;
+  };
+
+  async fetchCarto(query, format = 'json', method = 'get') {
+    const cleanedQuery = query.replace('\n', '');
+    const url = this.buildSqlUrl(cleanedQuery, format, method);
+
+    let fetchOptions = {};
+
+    if (method === 'post') {
+      fetchOptions = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+        },
+        body: `q=${cleanedQuery}&format=${format}`,
+      };
+    }
+
+    return fetch(url, fetchOptions)
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        throw new Error('Not found');
+      })
+      .then((d) => { // eslint-disable-line
+        return format === 'json' ? d.rows : d;
+      });
```

Comment:
Can we change this to async/await rather than classic promise methods?

Reply:
Ahh good catch!! Will do